Core/TimberApi/ObjectSelectionSystem/PickObjectTool.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Timberborn.Common;
+using Timberborn.CoreUI;
+using Timberborn.EntitySystem;
+using Timberborn.InputSystem;
+using Timberborn.SelectionSystem;
+using Timberborn.ToolSystem;
+using UnityEngine;
+
+namespace TimberApi.ObjectSelectionSystem
+{
+    public class PickObjectTool : Tool, IInputProcessor
+    {
+        //private static readonly string CursorKey = "PickObjectCursor";
+        private readonly InputService _inputService;
+        private readonly ToolManager _toolManager;
+        private readonly Highlighter _highlighter;
+        private readonly Colors _colors;
+        private readonly EntityComponentRegistry _entityComponentRegistry;
+        //private readonly CursorService _cursorService;
+        private readonly SelectableObjectRaycaster _selectableObjectRaycaster;
+        private ToolDescription _toolDescription;
+        private string _warning;
+        private readonly HashSet<GameObject> _allCandidates = new HashSet<GameObject>();
+        private Func<GameObject, string> _validateCandidate;
+        private Action<GameObject> _callback;
+
+        public PickObjectTool(InputService inputService, ToolManager toolManager, Highlighter highlighter, Colors colors, EntityComponentRegistry entityComponentRegistry, CursorService cursorService, SelectableObjectRaycaster selectableObjectRaycaster)
+        {
+            _inputService = inputService;
+            _toolManager = toolManager;
+            _highlighter = highlighter;
+            _colors = colors;
+            _entityComponentRegistry = entityComponentRegistry;
+            //_cursorService = cursorService;
```

Comment:
Can comment out code be removed ?

Reply:
Added a comment why commented code is there